api/src/lib/form-handler.js

```diff
@@ -6,6 +6,9 @@ module.exports = (
   clientService
 ) => {
   return (templateName, getView, postHandler) => async (request, reply, source, error) => {
+    if (error && error.output.statusCode === 404) {
+      reply(error);
+    }
```

Comment:
I think you'll probably need to throw the rest of this function inside the `else` block of this if statement. Calling `reply` doesn't stop execution of the function